executor/prepared.go

```diff
@@ -339,20 +339,20 @@ func ResetStmtCtx(ctx context.Context, s ast.StmtNode) {
 	case *ast.UpdateStmt:
 		sc.IgnoreTruncate = false
 		sc.OverflowAsWarning = false
-		sc.TruncateAsWarning = !sessVars.StrictSQLMode
+		sc.TruncateAsWarning = !sessVars.StrictSQLMode || stmt.IgnoreErr
 		sc.InUpdateOrDeleteStmt = true
-		sc.IgnoreError = stmt.IgnoreErr
+		sc.DividedByZeroAsWarning = stmt.IgnoreErr
```

Comment:
should we take SQLMode into consideration here?

Reply:
I handle SQLMode in `handleDivisionByZeroError` now. Your suggestion is move some process from `handleDivisionByZeroError` to here?